RNiTunes/RNiTunes.m

```diff
@@ -196,7 +205,15 @@ @implementation RNiTunes
         NSDictionary *songDictionary = [NSMutableDictionary dictionary];
 
         if (fields == nil) {
-            songDictionary = @{@"albumTitle":albumTitle, @"albumArtist": albumArtist, @"duration":[duration isKindOfClass:[NSString class]] ? [NSNumber numberWithInt:[duration intValue]] : duration, @"genre":genre, @"playCount": [NSNumber numberWithInt:[playCount intValue]], @"title": title};
+            songDictionary = @{
+              @"albumTitle": albumTitle,
+              @"albumArtist": albumArtist,
+              @"assetUrl": assetUrl,
+              @"duration": [duration isKindOfClass:[NSString class]] ? [NSNumber numberWithInt:[duration intValue]] : duration,
```

Comment:
Don't you know what the duration is, though? You define/load it above. Edit: Oh this isn't your code.